### PR TITLE
Fix #507 safari converter generichide

### LINKED ISSUE #507: @@*$generichide issue
Generichide exceptions don't work correctly in Safari.

STR:
1. Install 2.5.11 extension to Safari
2. Import [rules.txt](https://github.com/AdguardTeam/AdguardBrowserExtension/files/694100/rules.txt)
3. Open:
www.championat.com
www.khabar.ndtv.com
www.bild.de

### PR DESCRIPTION
**Pull request by Mizzick**
#507

### CODE REVIEW

File: Extension/browser/safari/lib/converter.js [hunk 1]
```diff
@@ -307,7 +307,8 @@ var SafariContentBlockerConverter = {
 
             function isUrlBlockRule(r) {
                 return self._isContentType(r, adguard.rules.UrlFilterRule.contentTypes.URLBLOCK) ||
-                    self._isContentType(r, adguard.rules.UrlFilterRule.contentTypes.GENERICBLOCK);
+                    self._isContentType(r, adguard.rules.UrlFilterRule.contentTypes.GENERICBLOCK) ||
```
Comment (ameshkov) [inline Extension/browser/safari/lib/converter.js]: I don't want to get rid of `if-document`.

We should first find out if it is possible to use `url-filter`

File: Extension/tests/safari-converter/safari-correct-rules.js [hunk 2]
```diff
@@ -35,10 +35,13 @@ var safariCorrectRules = [
   },
   {
     "trigger": {
-      "url-filter": "^https?://([^/]*\\.)?test-elemhide\\.com",
+      "url-filter": ".*",
       "resource-type": [
```
Comment (ameshkov) [inline Extension/tests/safari-converter/safari-correct-rules.js]: We don't need `resource-type` anymore

File: Extension/browser/safari/lib/converter.js [hunk 3]
```diff
@@ -311,15 +311,26 @@ var SafariContentBlockerConverter = {
                     self._isContentType(r, adguard.rules.UrlFilterRule.contentTypes.GENERICBLOCK);
             }
 
+            function isCssHideRule(r) {
```
Comment (ameshkov) [inline Extension/browser/safari/lib/converter.js]: It'd be better to rename it to `isCssExceptionRule`

**Comment by Mizzick** [comment]
Fixed

File: Extension/browser/safari/lib/converter.js [hunk 4]
```diff
@@ -829,6 +823,7 @@ var SafariContentBlockerConverter = {
 
         this._applyDomainWildcards(converted);
         adguard.console.info('Content blocker length: ' + converted.length);
+        console.log(contentBlocker.errors);
```
Comment (ameshkov) [inline Extension/browser/safari/lib/converter.js]: console.log!

**Pull request merged**